sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpEventHubClient.cs

```diff
@@ -479,7 +485,9 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
             }
             catch (Exception ex)
             {
+                _closed = false;
```

Comment:
Can we still use the client in case a `TaskCanceledException` is thrown right after we close the management link? Just wanted to make sure it's not left in a weird state.

Reply:
Same question for producer and consumer.

Reply:
Good point.   For some reason, I had this in the client but not in the consumer or producer.   I shifted things so that the check happens before we close the link.   Once we close it, the resulting flow is synchronous and in a contained scope, so we should allow it to finish rather than preempting for cancellation.

Adjustment made to the client, producer, and consumer.